sw/device/silicon_creator/lib/cert/dice_cwt.c

```diff
@@ -5,29 +5,63 @@
 #include <stdint.h>
 
 #include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/silicon_creator/lib/base/util.h"
 #include "sw/device/silicon_creator/lib/cert/cbor.h"
 #include "sw/device/silicon_creator/lib/cert/cert.h"
 #include "sw/device/silicon_creator/lib/cert/dice.h"
 #include "sw/device/silicon_creator/lib/drivers/hmac.h"
 #include "sw/device/silicon_creator/lib/drivers/keymgr.h"
 #include "sw/device/silicon_creator/lib/drivers/otp.h"
 #include "sw/device/silicon_creator/lib/error.h"
+#include "sw/device/silicon_creator/lib/otbn_boot_services.h"
 #include "sw/device/silicon_creator/lib/sigverify/ecdsa_p256_key.h"
 #include "sw/device/silicon_creator/manuf/lib/flash_info_fields.h"
 
 #include "otp_ctrl_regs.h"  // Generated.
 
-// Keys
-const int64_t kCoseKeyKtyLabel = 1;
-const int64_t kCoseKeyAlgLabel = 3;
-const int64_t kCoseEc2CrvLabel = -1;
-const int64_t kCoseEc2XLabel = -2;
-const int64_t kCoseEc2YLabel = -3;
+// Labels
+#define CWT_COSE_KEY_LABEL_TYPE (1)
+#define CWT_COSE_KEY_LABEL_ALG (3)
+#define CWT_COSE_KEY_LABEL_CRUVE (-1)
+#define CWT_COSE_KEY_LABEL_EC2_X (-2)
+#define CWT_COSE_KEY_LABEL_EC2_Y (-3)
+#define CWT_LABEL_ISS (1)
+#define CWT_LABEL_SUB (2)
+#define CWT_LABEL_CODE_HASH (-4670545)
+#define CWT_LABEL_CFG_HASH (-4670547)
+#define CWT_LABEL_CFG_DESCR (-4670548)
+#define CWT_LABEL_AUTH_HASH (-4670549)
+#define CWT_LABEL_MODE (-4670551)
+#define CWT_LABEL_SUBJECT_PK (-4670552)
+#define CWT_LABEL_KEY_USAGE (-4670553)
+#define CWT_LABEL_PROFILE_NAME (-4670554)
 
 // Values
-const int64_t kCoseKeyAlgEcdsa256 = -7;
-const int64_t kCoseEc2CrvP256 = 1;
-const int64_t kCoseKeyKtyEc2 = 2;
+#define CWT_COSE_KEY_ALG_ECDSA256 (-7)
+#define CWT_COSE_EC2_CRUVE_P256 (1)
+#define CWT_COSE_KEY_TYPE_EC2 (2)
+
+#define CWT_PROFILE_NAME "android.16"
+#define CWT_PROFILE_NAME_VALUE_LEN 10 /* "android.16" */
+#define CWT_PROFILE_NAME_LEN (1 + CWT_PROFILE_NAME_VALUE_LEN)
+
+#define CWT_ISSU_SUBJ_KEY_ID_LEN kHmacDigestNumBytes
+#define CWT_ISSU_SUBJ_LEN (CWT_ISSU_SUBJ_KEY_ID_LEN * 2)
+
+// Entry numbers
+enum {
+  kCoseKeyMapCount = 5,
+  kDiceChainEntryArrayCount = 4,
+  kDiceChainEntryInputArrayCount = 4,
+  kDiceChainEntryPayloadMapCount = 10,
+};
+
+// CWT sizes
+enum {
+  kCoseKeyEcP256SizeBytes = 77,
+  kCdi0MaxPayloadSizeBytes = 448,
```

Comment:
is Cdi1 the same max size?

Reply:
No, it'll be different if we include owner_manifest_measurement in the CSR.